lib/embag.cc

```diff
@@ -250,7 +249,7 @@ bool Bag::readRecords(boost::iostreams::stream<T> &stream) {
    * earlier in the file. Each CHUNK_INFO knows the position of its corresponding chunk.
    */
   for (size_t i = 0; i < chunk_count; i++) {
-    const auto info = chunk_infos_[i];
+    auto& info = chunk_infos_[i];
```

Comment:
What's the reasoning behind this change?

Reply:
Resolve offline: We need to update the info.num_messages value as we move through the info block